src/test/java/seedu/address/logic/commands/ViewCommandTest.java

```diff
@@ -0,0 +1,126 @@
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.Person;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for
+ * {@code DeleteCommand}.
+ */
+public class ViewCommandTest {
+
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_validIndexUnfilteredList_success() {
+        Person personToView = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+        ViewCommand viewCommand = new ViewCommand(INDEX_FIRST_PERSON);
+
+        String expectedMessage = String.format(MESSAGE_VIEW_PERSON_SUCCESS,
+                Messages.format(personToView));
+
+        ModelManager expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModel.viewPerson(personToView);
+
+        assertCommandSuccess(viewCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidIndexUnfilteredList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredPersonList().size() + 1);
+        ViewCommand viewCommand = new ViewCommand(outOfBoundIndex);
+
+        assertCommandFailure(viewCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_invalidIndexFilteredList_throwsCommandException() {
+        showPersonAtIndex(model, INDEX_FIRST_PERSON);
+
+        Index outOfBoundIndex = INDEX_SECOND_PERSON;
+        // ensures that outOfBoundIndex is still in bounds of address book list
+        assertTrue(outOfBoundIndex.getZeroBased() < model.getAddressBook().getPersonList().size());
+
+        ViewCommand viewCommand = new ViewCommand(outOfBoundIndex);
+
+        assertCommandFailure(viewCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void equals() {
+        ViewCommand viewFirstCommand = new ViewCommand(INDEX_FIRST_PERSON);
+        ViewCommand viewSecondCommand = new ViewCommand(INDEX_SECOND_PERSON);
+
+        // same object -> returns true
+        assertEquals(viewFirstCommand, viewFirstCommand);
+
+        // same values -> returns true
+        ViewCommand viewFirstCommandCopy = new ViewCommand(INDEX_FIRST_PERSON);
+        assertEquals(viewFirstCommand, viewFirstCommandCopy);
+
+        // different types -> returns false
+        assertNotEquals(1, viewFirstCommand);
+
+        // null -> returns false
+        assertNotEquals(null, viewFirstCommand);
+
+        // different person -> returns false
+        assertNotEquals(viewFirstCommand, viewSecondCommand);
+    }
+
+    @Test
+    public void execute_emptyAddressBook_throwsCommandException() {
+        // Empty the address book
+        model = new ModelManager();
```

Comment:
If you end up doing the `setUp` method call, can check if you still need a plain ModelManager constructor with no params passed or need to pass in the UserPrefs and all